syntax = "proto3";

package protos;

import "common.proto";

message SegmentRequest {
  SegmentId segment_id = 1;
}

message SegmentResponse {
  Segment segment  = 1;
}

message SegmentsRequest {
  SegmentId segment_id = 1;
}

message SegmentsResponse {
  repeated SegmentId segments = 2;
}



message ConfigureRequest {
  repeated Table tables = 1;
}

message InsertRequest {
  string table = 1;
  Row row = 2;
}


service Packer {
  rpc Insert(InsertRequest) returns (Error) {}
  rpc Segment(SegmentRequest) returns (SegmentResponse) {}
  rpc Segments(SegmentsRequest) returns (SegmentsResponse) {}
}
